Jint/Runtime/Interpreter/Expressions/JintUnaryExpression.cs

```diff
@@ -165,7 +166,8 @@ protected override object EvaluateInternal()
                         }
                     }
 
-                    var v = _engine.GetValue(value, true);
+                    // TODO: double evaluation problem
```

Comment:
This is a problem that need more work to fix. GetValue is overloaded by different expressions, but we need to first check if it's an undefined reference so that we can return `"undefined"` without an error. At the moment this should pass all the tests, but it not the correct behaviour as you can try with this example:

```
const fn = () => Log('test');
typeof fn();
```

Reply:
So what's the correct behavior, this is what node gives:

```
Welcome to Node.js v14.16.1.
Type ".help" for more information.
> const fn = () => Log('test');
undefined
> typeof fn();
Uncaught ReferenceError: Log is not defined
    at fn (REPL1:1:18)
```

And Jint follows the logic:

```
jint> const fn = () => Log('test');
undefined
jint> typeof fn();
Jint.Runtime.JavaScriptException: Log is not defined
   at fn () repl:1:1
   at repl:1:8
```

Reply:
ye, of course, Log is not defined 😅, you have to use the corresponding print function.

With node:
```
Type ".help" for more information.
> const fn = () => { console.log('test'); }
undefined
> typeof fn();
test
'undefined'
```

With jint:
```
jint> const fn = () => { print('test'); }
undefined
jint> typeof fn();
test
test
"undefined"
jint>
```

Reply:
OK thanks, can you open a separate issue so that can be looked into, unless you want to solve all in this PR 😉